### PR TITLE
Improving Spread Attack

### PR DESCRIPTION
**Pull request by CodingSquirrel**
First Attacks are uniformly spread across targets. All further attacks
are random.

### CODE REVIEW

File: lua/spreadattack.lua [hunk 1]
```diff
@@ -132,6 +132,12 @@ function SpreadAttack()
       if beginAttack == nil or endAttack == beginAttack then
         break
       end
+      
+      local stableNum = math.floor((endAttack - beginAttack + 1) / table.getn(curSelection)) + ((math.mod(endAttack - beginAttack + 1, table.getn(curSelection)) >= index) and 1 or 0)
```
Comment (ChrisKitching) [inline lua/spreadattack.lua]: Perform common subexpression elimination.

File: lua/spreadattack.lua [hunk 2]
```diff
@@ -132,6 +132,12 @@ function SpreadAttack()
...
+      for i = 0, stableNum - 1 do
+        unitorders[i + beginAttack],unitorders[((i) * table.getn(curSelection)) + beginAttack + index - 1] = unitorders[((i) * table.getn(curSelection)) + beginAttack + index - 1],unitorders[i + beginAttack]
```
Comment (ChrisKitching) [inline lua/spreadattack.lua]: Likewise.

File: lua/spreadattack.lua (hunk 1 above)
Comment (ChrisKitching) [inline lua/spreadattack.lua]: Also, `and 1 or 0` is pretty obtuse ;)

Comment (CodingSquirrel) [inline lua/spreadattack.lua]: But that would make it legible.

Also, lack of ternary operator fail.

**Comment by CodingSquirrel** [comment]
Updated PR with cleaner logic.

**Comment by ChrisKitching** [comment]
> Updated PR with cleaner logic.

And some shiny shiny merge conflicts. Rebase please :)

**Pull request closed without merging**